File: Programmers/ParkJihyeon/정렬/Programmers_hIndex.java
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

public class Programmers_hIndex {
    public static class Solution {
        public int solution(int[] citations) {
            int answer = 0;
            int len = citations.length;

            Arrays.sort(citations);

            int i;
            int cnt = 0;
            int index = len - 1;

            for (i = len; i >= 0; i--) {
                while(citations[index] >= i){
                    cnt ++;
                    index --;
//                    System.out.println("cnt: " + cnt+ " " +"index:" + index);
                    if (index < 0) break;
                }
                if(cnt >= i) break;
            }
            answer = i;
            return answer;
        }

        public static void main(String[] args) {
            int[] citations = {3, 0, 6, 1, 5};
            System.out.print(new Programmers_hIndex.Solution().solution(citations));
        }
    }
}

File: Programmers/HaJeongho/정렬/Programmers_hIndex.py
def solution(citations):
    answer = 0
    for i in range(1, len(citations)):
        while i!=0:
            if citations[i]<citations[i-1]:
                citations[i],citations[i-1] = citations[i-1], citations[i]
                i -= 1
            else:
                break
    for i in range(citations[-1]):
        if citations[-i]>=i:
            answer = i
        else:
            break
    return answer
